docs/src/pages/guides/typescript/typescript.md

```diff
@@ -2,7 +2,7 @@
 
 <p class="description">You can add static typing to JavaScript to improve developer productivity and code quality thanks to TypeScript.</p>
 
-Have a look at the [Create React App with TypeScript](https://github.com/mui-org/material-ui/tree/master/examples/create-react-app-with-typescript) example. A minimum version of TypeScript 2.8 is required.
+Have a look at the [Create React App with TypeScript](https://github.com/mui-org/material-ui/tree/master/examples/create-react-app-with-typescript) example. A minimum version of TypeScript 3.2 is required.
```

Comment:
Or perhaps using the wording from https://material-ui.com/getting-started/supported-platforms/#react:

```suggestion
Material-UI supports the most recent versions of TypeScript, starting with ^3.2.1.

Have a look at the [Create React App with TypeScript](https://github.com/mui-org/material-ui/tree/master/examples/create-react-app-with-typescript) example.
```

I picked `3.2.1` due to 
![image](https://user-images.githubusercontent.com/2852660/82860854-d3cee200-9f12-11ea-9624-fd4f660fe8fd.png) but I guess it doesn't matter hugely.